numba/np/arrayobj.py

```diff
@@ -5266,3 +5264,31 @@ def impl(arr):
                        "is ambiguous. Use a.any() or a.all()")
                 raise ValueError(msg)
         return impl
+
+
+@overload(np.swapaxes)
+def numpy_swapaxes(arr, axis1, axis2):
+    if not isinstance(axis1, (int, types.Integer)):
+        raise errors.TypingError('The second argument "axis1" must be an '
+                                 'integer')
+    if not isinstance(axis2, (int, types.Integer)):
+        raise errors.TypingError('The third argument "axis2" must be an '
+                                 'integer')
+    if not isinstance(arr, types.Array):
+        raise errors.TypingError('The first argument "arr" must be an array')
```

Comment:
I think that this is permissible given the reliance on `ndim` later, but NumPy supports anything array-like, e.g.:
```python
In [6]: np.swapaxes((1, 2), 0, 0)
Out[6]: array([1, 2])
```

Reply:
This is true, but I was not able to figure out how to solve it without generating dynamically a tuple. Is it sufficient to mention this in the documentation or does it need to be fixed?

Reply:
I think it's fine to leave for now in the interests of completing the feature. The compiler will catch invalid types and if its really important to someone it's likely they'll open an issue.